finalfrontier/src/vocab.rs

```diff
@@ -456,6 +534,30 @@ fn bracket(word: &str) -> String {
     bracketed
 }
 
+fn search_first_occurrence<S>(vec: &[CountedType<S>], key: &CountedType<S>) -> Result<usize, Error>
```

Comment:
Do not implement binary search by hand, use `binary_search_*`. See above for a suggested approach.